packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -512,6 +512,123 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+
+      await tester.tap(find.byKey(childKey));
+      await tester.pump();
+
+      await expectLater(
+        find.byKey(repaintBoundaryKey),
+        matchesGoldenFile('inspector.overlay_positioning_with_transform.png'),
+      );
+    });
+
+    testWidgets('Multiple widget inspectors', (WidgetTester tester) async {
+      // This test verifies that interacting with different inspectors
+      // works correctly. This use case may be an app that displays multiple
+      // apps inside (i.e. a storyboard).
+      final GlobalKey selectButton1Key = GlobalKey();
+      final GlobalKey selectButton2Key = GlobalKey();
+
+      final GlobalKey inspector1Key = GlobalKey();
+      final GlobalKey inspector2Key = GlobalKey();
+
+      final GlobalKey child1Key = GlobalKey();
+      final GlobalKey child2Key = GlobalKey();
+
+      InspectorSelectButtonBuilder selectButtonBuilder(Key key) {
+        return (BuildContext context, VoidCallback onPressed) {
+          return Material(child: RaisedButton(onPressed: onPressed, key: key));
+        };
+      }
+
+      // State type is private, hence using dynamic.
+      // The inspector state is static, so it's enough with reading one of them.
+      dynamic getInspectorState() => inspector1Key.currentState;
+      String paragraphText(RenderParagraph paragraph) {
+        final TextSpan textSpan = paragraph.text as TextSpan;
+        return textSpan.text;
+      }
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: Row(
+            children: <Widget>[
+              Flexible(
+                child: WidgetInspector(
+                  key: inspector1Key,
+                  selectButtonBuilder: selectButtonBuilder(selectButton1Key),
+                  child: Container(
+                    key: child1Key,
+                    child: const Text('Child 1'),
+                  ),
+                ),
+              ),
+              Flexible(
+                child: WidgetInspector(
+                  key: inspector2Key,
+                  selectButtonBuilder: selectButtonBuilder(selectButton2Key),
+                  child: Container(
+                    key: child2Key,
+                    child: const Text('Child 2'),
+                  ),
+                ),
+              ),
+            ],
+          ),
+        ),
+      );
+
+      final InspectorSelection selection = getInspectorState().selection as InspectorSelection;
+      // The selection is static, so it may be initialized from previous tests.
+      selection?.clear();
+
+      await tester.tap(find.text('Child 1'));
+      await tester.pump();
+      expect(paragraphText(selection.current as RenderParagraph), equals('Child 1'));
+
+      await tester.tap(find.text('Child 2'));
```

Comment:
wow I'm surprised this multiple WidgetInspector case works this well :)

Reply:
Very niche use case I would say, but I was surprised as well to see it working perfectly fine in a sample app where I took it to the extreme. Multiple MaterialApps and even nesting between them ^^